builder/builder_utilities.livecodescript

```diff
@@ -555,42 +589,54 @@ private function builderMergExtDownloadedFilePath pMergExtVersion, pEdition
-   if there is not a file tZipPath then
-      local tUrl
-      put builderMergExtUrl(pMergExtVersion, pEdition) into tUrl
-      builderLog "message", "Downloading MergExt to" && tZipPath
-      get shell(merge("curl -o '[[tZipPath]]' '[[tUrl]]'"))
-      if the result is not zero then
-         builderLog "error", "Failed to download MergExt zip:" && it
-         throw "failure"
-      end if
-   end if
-end builderMergExtEnsureZip
+private command builderMergExtUnpack pMergExtVersion, pEdition
+   local tFolder
+   put builderUnpackFolder() & slash & "Ext" into tFolder
+   
+   local tZipFile
+   put builderMergExtDownloadedFilePath(pMergExtVersion, pEdition) into tZipFile
+   
+   local tURL
+   put builderMergExtUrl(pMergExtVersion, pEdition) into tURL
+   
+   builderEnsureZip tZipFile, tURL
+   
+   builderUnpackZip tZipFile, tFolder
+end builderMergExtUnpack
 
-command builderMergExtUnpack pMergExtVersion, pEdition
-   builderMergExtEnsureZip pMergExtVersion, pEdition
+////////////////////////////////////////////////////////////////////////////////
+
+private function builderTSNetUrl pTSNetVersion
+   return "https://downloads.livecode.com/tsNet/tsNet_" & pTSNetVersion & ".zip"
+end builderTSNetUrl
+
+private function builderTSNetDownloadedFilePath pTSNetVersion
+   return builderRepoFolder() & slash & "tsNet_" & pTSNetVersion & ".zip"
+end builderTSNetDownloadedFilePath
+
+private command builderTSNetUnpack pTSNetVersion
+   local tFolder
+   put builderUnpackFolder() & slash & "Ext" & slash & "tsNet" into tFolder
    
-   local tMergExtFolder
-   local tMergExtZip
-   put builderMergExtFolder() into tMergExtFolder
-   put builderMergExtDownloadedFilePath(pMergExtVersion, pEdition) into tMergExtZip
+   local tZipFile
+   put builderTSNetDownloadedFilePath(pTSNetVersion) into tZipFile
    
-   if there is a folder tMergExtFolder then
-      get shell(merge("rm -r '[[tMergExtFolder]]'"))
-   end if
+   local tURL
+   put builderTSNetUrl(pTSNetVersion) into tURL
    
-   get shell(merge("mkdir -p '[[tMergExtFolder]]/MergExt'"))
-   if the result is not zero then
-      builderLog "error", "Failed to create MergExt folder:" && it
-      throw "failure"
-   end if
+   builderEnsureZip tZipFile, tURL
    
-   get shell(merge("cd '[[tMergExtFolder]]/MergExt' && unzip '[[tMergExtZip]]'"))
-   if the result is not zero then
-      builderLog "error", "Failed to unzip MergExt bundle:" && it
-      throw "failure"
+   builderUnpackZip tZipFile, tFolder
+end builderTSNetUnpack
+
+////////////////////////////////////////////////////////////////////////////////
+
+command builderExtUnpack pEdition
+   -- Ensure that the MergExt bundle is downloaded and unpacked
+   builderMergExtUnpack builderMergExtVersion(), (toUpper(char 1 of pEdition) & toLower(char 2 to -1 of pEdition))
+   
+   if pEdition is "community" then
+      -- Ensure that the TSNet bundle is downloaded and unpacked
+      builderTSNetUnpack builderTSNetVersion()
```

Comment:
@montegoulding That's cool, I figured you were still working on it.  I was just having a quick look over it and noticed those two things, so thought I'd highlight them for you!